src/components/Disturber.tsx

```diff
@@ -0,0 +1,170 @@
+import { DrawerNavigationProp } from '@react-navigation/drawer';
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+
+import { Icon, colors, normalize } from '../config';
+import { addToStore, findClosestItem, isActive, readFromStore } from '../helpers';
+import { useHomeRefresh, useStaticContent } from '../hooks';
+
+import { Button } from './Button';
+import { Image } from './Image';
+import { BoldText, RegularText } from './Text';
+import { Wrapper, WrapperHorizontal } from './Wrapper';
+
+type Props = {
+  navigation: DrawerNavigationProp<any>;
+  name: string;
+};
+
+interface DateRange {
+  dateEnd: string;
+  dateStart: string;
+}
+
+interface DataItem {
+  backgroundColor?: string;
+  button: {
+    navigationTo: string;
+    params: { title?: string; webUrl: string };
+    title?: string;
+  };
+  category?: string;
+  dates: DateRange[];
+  description: string;
+  id: number;
+  picture?: { aspectRatio?: { HEIGHT: number; WIDTH: number }; uri: string };
+  title: string;
+}
+
+// eslint-disable-next-line complexity
+export const Disturber = ({ navigation, name }: Props) => {
+  const [isVisible, setIsVisiblle] = useState(false);
+  const refreshTimeKey = `publicJsonFile-${name}`;
+
+  const { data, refetch } = useStaticContent<DataItem[]>({
+    refreshTimeKey,
+    name,
+    type: 'json'
+  });
+
+  useHomeRefresh(refetch);
+
+  // filter data for present items and items with active date/time periods
+  const isActiveData = data?.filter((item) => item && isActive(item));
+
+  // find the closest item to the current date/time
+  const closestItem: DataItem | null = findClosestItem(isActiveData || []);
+
+  const setDistruberComplete = () => {
+    setIsVisiblle((prev) => !prev);
+    addToStore(name, closestItem?.id.toString());
+  };
+
+  useEffect(() => {
+    readFromStore(name).then((value) => {
+      if (closestItem?.id && value !== closestItem?.id.toString()) setIsVisiblle(true);
+    });
+  }, [closestItem]);
+
+  if (!isVisible || !closestItem) return null;
+
+  return (
+    <Wrapper style={{ padding: normalize(24) }}>
```

Comment:
can the wrapper style be excluded in styles at the bottom, like in `<Wrapper style={styles.noPaddingTop}>` or at some additional prop in `Wrapper`?

Reply:
Yeah, you're absolutely right. I missed it. I'll fix it.